packages/react-client/src/devices/mediaInitializer.ts

```diff
@@ -6,27 +6,41 @@ import { prepareConstraints } from "./constraints";
 type AudioVideo<T> = { audio: T; video: T };
 
 type MediaConstraints = AudioVideo<MediaTrackConstraints | undefined | boolean>;
-
 type PreviousDevices = AudioVideo<MediaDeviceInfo | null>;
 
 const defaultErrors = { audio: null, video: null };
 
+const errorMap: Record<string, DeviceError> = {
+  NotFoundError: NOT_FOUND_ERROR,
+  OverconstrainedError: OVERCONSTRAINED_ERROR,
+  NotAllowedError: PERMISSION_DENIED,
+};
+
+const getSingleMedia = async <T extends "audio" | "video">(
+  type: T,
+  constraints: MediaStreamConstraints[T],
+): Promise<[MediaStream, null] | [null, DeviceError]> => {
+  const baseConstraints = { audio: undefined, video: undefined };
+
+  try {
+    const stream = await navigator.mediaDevices.getUserMedia({ ...baseConstraints, [type]: constraints });
+    return [stream, null];
+  } catch (err) {
+    if (!(err instanceof DOMException)) return [null, UNHANDLED_ERROR];
+
+    return [null, errorMap[err.name] ?? UNHANDLED_ERROR];
```

Comment:
I would avoid negation in condition here, for me it makes code easier to read:
```suggestion
    if (err instanceof DOMException) {
      return [null, errorMap[err.name] ?? UNHANDLED_ERROR];
    }

    return [null, UNHANDLED_ERROR];
```

Reply:
👀

Reply:
this will get in the PR, i think i stashed it